src/Algebra/Properties/Magma.agda

```diff
@@ -0,0 +1,74 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some items and theory for Magma
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (Magma)
+open import Algebra.Definitions using (Commutative)
+open import Data.Product using (_×_; _,_; ∃)
+open import Function using (_∘_)
+open import Level using (_⊔_)
+open import Relation.Binary using (Rel; _Respects_; _Respects₂_)
+open import Relation.Nullary using (¬_)
+
+module Algebra.Properties.Magma {a ℓ} (M : Magma a ℓ) where
+
+open Magma M
+
+------------------------------------------------------------------------------
+-- The divisibility relation
+------------------------------------------------------------------------------
+
+infix 5 _∣_ _∤_ _∣∣_ _¬∣∣_
+
+
+_∣_ : Rel Carrier (a ⊔ ℓ)          -- x ∣ y  denotes  x divides y
+x ∣ y = ∃ (\q → y ≈ (q ∙ x))
+
+_∤_ : Rel Carrier (a ⊔ ℓ)
+_∤_ x = ¬_ ∘ _∣_ x
+
+_∣∣_ : Rel Carrier (a ⊔ ℓ)
+x ∣∣ y = x ∣ y × y ∣ x
+
+-- _∣∣_ is mutual divisibility.
+-- The elements related by _∣∣_ are called associated  (when in a cancellative
+-- monoid).
+-- For a cancellative monoid, it follows from
+-- x ∣∣ y  that x and y differ only in a certain invertible factor.
+-- It will be proved further that  gcd a b  is unique, in the sense that
+-- g g' : GCD a b → g ∣∣ g'.
+
+_¬∣∣_ : Rel Carrier (a ⊔ ℓ)
+_¬∣∣_ x =  ¬_ ∘ _∣∣_ x
+
+------------------------------------------------------------------------------
+-- Properties of divisibility
+------------------------------------------------------------------------------
+
+∣-respˡ : ∀ {y} → (_∣ y) Respects _≈_
+∣-respˡ x≈x' (q , y≈qx) = q , trans y≈qx (∙-congˡ x≈x')
+
+∣-respʳ : ∀ {x} → (x ∣_) Respects _≈_
+∣-respʳ y≈y' (q , y≈qx) = q , trans (sym y≈y') y≈qx
+
+∣-resp : _∣_ Respects₂ _≈_
+∣-resp = (λ {x y y'} → ∣-respʳ {x} {y} {y'}) , (λ {y x x'} → ∣-respˡ {y} {x} {x'})
```

Comment:
Not sure why you're not simply writing `∣-respʳ, ∣-respˡ`?

Reply:
Done.